include/eve/module/core/regular/last_true.hpp

```diff
@@ -11,12 +11,87 @@
 
 namespace eve
 {
-EVE_MAKE_CALLABLE(last_true_, last_true);
+  template<typename Options>
+  struct last_true_t : conditional_callable<last_true_t, Options>
+  {
+    template<relaxed_logical_value T>
+    EVE_FORCEINLINE std::optional<std::ptrdiff_t> operator()(T v) const noexcept
+    {
+      static_assert(detail::validate_mask_for<decltype(this->options()), T>(),
+        "[eve::last_true] - Cannot use a relative conditional expression or a simd value to mask a scalar value");
+
+      return EVE_DISPATCH_CALL(v);
+    }
+
+    template<logical_simd_value T>
+    EVE_FORCEINLINE std::optional<std::ptrdiff_t> operator()(top_bits<T> v) const noexcept
+    {
+      return EVE_DISPATCH_CALL(v);
+    }
+
+    EVE_CALLABLE_OBJECT(last_true_t, last_true_);
+  };
+
+  //================================================================================================
+  //! @addtogroup core_reduction
+  //! @{
+  //!   @var last_true
+  //!   @brief Returns the index of the last element in the input which evaluates to `true`, if there is one.
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   @groupheader{Callable Signatures}
+  //!
+  //!   @code
+  //!   template <relaxed_logical_value L>
+  //!   std::optional<std::ptrdiff_t> last_true(L m) noexcept;                                               // 1
+  //!
+  //!   template <logical_simd_value L>
+  //!   std::optional<std::ptrdiff_t> last_true(top_bits<L> m) noexcept;                                     // 1
+  //!
+  //!   // lane masking
+  //!   std::optional<std::ptrdiff_t> last_true[conditional_expr auto c](/* any of the above */) noexcept;   // 2
+  //!   std::optional<std::ptrdiff_t> last_true[logical_value auto c](/* any of the above */) noexcept;      // 2
```

Comment:
NOTE: these comments are not correct for scalar values but I think they are OK regardless, yes conditionals don't work on scalars but we shouldn't just push it in ppl's faces everywhere.